src/accapi/client.py

```diff
@@ -160,48 +151,47 @@ def _receive_realtime_car_udpate(self):
         else:
             self._request_entry_list()
 
-    def _receive_entry_list(self):
-        entryList = EntryList.receive(self._receive)
+    def _receive_entry_list(self, msg):
+        entryList = EntryList.from_message(msg)
         self._cars = {i: self._cars[i] if i in self._cars else -1 for i in entryList.carIndices}
 
-    def _receive_entry_list_car(self):
-        args = EntryListCar.receive_args(self._receive)
+    def _receive_entry_list_car(self, msg):
+        args = EntryListCar.parse(msg)
         car = EntryListCar(*args)
         self._cars[car.carIndex] = len(car.drivers)
         for callback in self._onEntryListCarUpdate.callbacks:
             car = EntryListCar(*args)
             callback(Event(self, car = car))
 
-    def _receive_track_data(self):
-        args = TrackData.receive_args(self._receive)
+    def _receive_track_data(self, msg):
+        args = TrackData.parse(msg)
         for callback in self._onTrackDataUpdate.callbacks:
             data = TrackData(*args)
             callback(Event(self, data = data))
 
-    def _receive_broadcasting_event(self):
-        args = BroadcastingEvent.receive_args(self._receive)
+    def _receive_broadcasting_event(self, msg):
+        args = BroadcastingEvent.parse(msg)
         for callback in self._onBroadcastingEvent.callbacks:
             event = BroadcastingEvent(*args)
             callback(Event(self, event = event))
 
-    def _request_connection(
-        self,
-        displayName: str,
-        password: str,
-        updateIntervalMs: int,
-        commandPassword: str
-    ):
+    def _request_connection(self):
         self._send(
             ("B", OutboundMessageTypes.REGISTER_COMMAND_APPLICATION.value),
             ("B", self._broadcastingProtocolVersion),
-            ("s", displayName),
-            ("s", password),
-            ("i", self._updateIntervalMs),
-            ("s", commandPassword)
+            ("s", self._displayName),
+            ("s", self._password),
+            ("B", self._updateIntervalMs),
+            ("s", self._commandPassword),
+            # This padding seems to be needed, otherwise the game crashes (especially in multiplayer)
+            ("L", 0)
         )
 
     def _request_disconnection(self):
-        self._send(("B", OutboundMessageTypes.UNREGISTER_COMMAND_APPLICATION.value))
+        self._send(
+            ("B", OutboundMessageTypes.UNREGISTER_COMMAND_APPLICATION.value),
+            ("i", self._connectionId)
```

Comment:
I have not found documentation that says this is necessary, but I did add it to #4 as is. I have not had the opportunity to properly test the outcome of this request.